Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM --platform=$BUILDPLATFORM ubuntu as build
+ENV HOME="/root"
+WORKDIR $HOME
+
+RUN apt update && apt install -y build-essential curl python3-venv
+
+# Setup zig as cross compiling linker
+RUN python3 -m venv $HOME/.venv
+RUN .venv/bin/pip install cargo-zigbuild
```

Comment:
Hello @konstin , I was just looking through `ruff` commit history and saw this PR.

One comment here is usually one just uses system Python in a Docker container, mainly cuz Docker containers are short-lived objects already.

I wanted to suggest a nice `Dockerfile` linter [hadolint](https://github.com/hadolint/hadolint) that has a lot of good suggestions:
- Using `pip install --no-cache-dir` to minimize image size
- Using `apt-get` over `apt` and removing the apt cache when done

Here is a sample `pre-commit` integration:

```yaml
    - repo: https://github.com/hadolint/hadolint
      rev: v2.12.1-beta
      hooks:
          - id: hadolint-docker
```

Feel free to disregard this as well, just sharing for reference. Cheers!

Reply:
> One comment here is usually one just uses system Python in a Docker container, mainly cuz Docker containers are short-lived objects already.

I have bad experiences with user packages colliding with system packages, so i won't support that. As `pip install cargo-zigbuild` in the system environment will tell you: "Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv"

> Using pip install --no-cache-dir to minimize image size

This is a multi-stage docker build, the builder does not affect the final image size. As shown in the PR description, the total is 16.2MB, which is the size of the x86_64-unknown-linux-musl release binary.

> Using apt-get over apt and removing the apt cache when done

The `apt` interface is stable nowadays and the same as for the pip cache applies here.

Reply:
Alright sounds good!  The multi-stage build you set up here is pretty clever, nice work there